lib/helpers.bash

```diff
@@ -849,3 +849,30 @@ then
     fi
   }
 fi
+
+# `_bash-it-find-in-ancestor` uses the shell's ability to run a function in
+# a subshell to simplify our search to a simple `cd ..` and `[[ -r $1 ]]`
+# without any external dependencies. Let the shell do what it's good at.
+function _bash-it-find-in-ancestor() (
```

Comment:
>  this function creates it's own subshell by providing the command list within `(...)`

TIL !

So turns out this syntactical sugar and is the same as `function f() { ( ... ) }` :
```
$ function foo() ( echo foo; )
$ type foo
foo is a function
foo () 
{ 
    ( echo foo )
}
```

It also seems that the `{(...)}` technique is recommend to avoid confusing users who read the source code since its used so rarely.

**HOWEVER**

Since the calls sits are *already* sub-shells:
```
result="$(_bash-it-find-in-ancestor "${file}")"
```

Does making the function a sub-shell actually bring much value?

Wondering if a simple usage warning will suffice?

Reply:
> Wondering if a simple usage warning will suffice?

I expressly seek to avoid needing a usage warning. One possible reasonable example:
```
if _bash-it-find-in-ancestor ".hg" >/dev/null; then
  >&2 echo "This utility cannot operate from within a mercurial repository."
fi
```